tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/async/JaxRsClientAsyncTest.java

```diff
@@ -92,16 +94,60 @@ void setUp() {
 
     @Test
     public void testIntegrationWithJaxRsClient() throws Exception {
-        basicClient.get("/JaxRsClientAsyncTestEndpoint/jaxrsclient");
+        basicClient.get("/JaxRsClientAsyncTestEndpoint/jaxrsclient?baggageValue=" + QUERY_VALUE);
         readSpans();
     }
 
     @Test
     public void testIntegrationWithJaxRsClientAsync() throws Exception {
-        basicClient.get("/JaxRsClientAsyncTestEndpoint/jaxrsclientasync");
+        basicClient.get("/JaxRsClientAsyncTestEndpoint/jaxrsclientasync?baggageValue=" + QUERY_VALUE);
         readSpans();
     }
 
+    @Test
+    public void testIntegrationWithJaxRsClientError() throws Exception {
+        basicClient.get("/JaxRsClientAsyncTestEndpoint/jaxrsclienterror");
+        readErrorSpans();
+    }
+
+    public void readErrorSpans() {
+
+        List<SpanData> spanData = spanExporter.getFinishedSpanItems(3);
+
+        List<SpanData> serverSpans = spanExporter.getSpansWithKind(SpanKind.SERVER);
+
+        SpanData firstURL = null;
+        SpanData secondURL = null;
+        for (SpanData span : serverSpans) {
+            if (span.getAttributes().get(HTTP_TARGET).contains("JaxRsClientAsyncTestEndpoint/jaxrsclient")) {
+                firstURL = span;
+            } else {
+                secondURL = span;
+            }
+        }
+
+        Assert.assertNotNull(firstURL);
+        Assert.assertNotNull(secondURL);
+
+        SpanData httpGet = spanExporter.getFirst(SpanKind.CLIENT);
+
+        // Assert correct parent-child links
+        // Shows that propagation occurred
+        Assert.assertEquals(httpGet.getSpanId(), secondURL.getParentSpanId());
+        Assert.assertEquals(firstURL.getSpanId(), httpGet.getParentSpanId());
+
+        Assert.assertEquals(HTTP_INTERNAL_ERROR, firstURL.getAttributes().get(HTTP_STATUS_CODE).intValue());
+        Assert.assertEquals(HttpMethod.GET, firstURL.getAttributes().get(HTTP_METHOD));
+        Assert.assertEquals("http", firstURL.getAttributes().get(HTTP_SCHEME));
+
+        // There are many different URLs that will end up here. But all should contain "JaxRsClientAsyncTestEndpoint"
+        Assert.assertTrue(httpGet.getAttributes().get(HTTP_URL).contains("JaxRsClientAsyncTestEndpoint"));
+
+        Assert.assertEquals(HTTP_INTERNAL_ERROR, httpGet.getAttributes().get(HTTP_STATUS_CODE).intValue());
+        Assert.assertEquals(HttpMethod.GET, httpGet.getAttributes().get(HTTP_METHOD));
+        Assert.assertTrue(httpGet.getAttributes().get(HTTP_URL).contains("JaxRsClientAsyncTestEndpoint"));
```

Comment:
Add comments explaining why we're expecting the given status codes.

I think the flow is server receives request (`firstURL`), makes client request (`httpGet`), received by server (`secondURL`) which returns a 500 status code, which is received by the client causing it to throw an exception which is thrown out of the resource method, causing the first server request to return a 500 as well?

Should we also assert the attributes on `secondURL` here?